hack/update-codegen.sh

```diff
@@ -50,14 +58,28 @@ function generate_group() {
     --output-package ${INFORMERS_PKG} \
     --output-base ${SCRIPT_BASE}
 
+  # The following generators do not respect the --output-package, so generate
+  # their output into a temporary location and rsync it over.
+  local GEN_TMPDIR="$(mktemp -d /tmp/genXXXXXXXX)"
+
   echo "generating deep copies"
-  bazel run @io_k8s_code_generator//cmd/deepcopy-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION}
+  bazel run @io_k8s_code_generator//cmd/deepcopy-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION} --output-base ${GEN_TMPDIR}
 
   echo "generating defaults"
-  bazel run @io_k8s_code_generator//cmd/defaulter-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION}
+  bazel run @io_k8s_code_generator//cmd/defaulter-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION} --output-base ${GEN_TMPDIR}
 
   echo "generating conversions"
-  bazel run @io_k8s_code_generator//cmd/conversion-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION}
+  bazel run @io_k8s_code_generator//cmd/conversion-gen -- --input-dirs ${APIS_PKG}/${GROUP_NAME} --input-dirs ${APIS_PKG}/${GROUP_NAME}/${VERSION} --output-base ${GEN_TMPDIR}
+
+  rsync -a "${GEN_TMPDIR}/k8s.io/cluster-registry/" "${SCRIPT_BASE}/${SCRIPT_PACKAGE}"
+
+  if [[ "${GEN_TMPDIR}" == "/tmp/gen"* ]]; then
+    rm -rf "${GEN_TMPDIR}"
+  fi
 }
 
+echo "Creating temporary GOPATH from bazel workspace"
+TMP_GOPATH="$("${SCRIPT_ROOT}/gopath_from_workspace.sh")"
+trap cleanup EXIT
```

Comment:
s/cleanup/clean_up/g

Hit this running locally:

```
$ cd $( mktemp -d )
$ GOPATH=$PWD
$ git clone https://github.com/perotinus/cluster-registry.git src/k8s.io/cluster-registry
$ cd src/k8s.io/cluster-registry
$ git checkout useworkspacereposforgen
$ bazel clean --expunge
$ ./hack/update-codegen.sh
# lot of output
./hack/update-codegen.sh: line 1: cleanup: command not found   
```

Reply:
Thanks, fixed!